### ISSUE TITLE
Confusion regarding output and sigmoid

### ISSUE THREAD

**Issue by bibhashthakur** [jocicmarko/ultrasound-nerve-segmentation#23]
I am using the Dice loss function you have stated in my code. 
The question I have is that you're using sigmoid as the activation function in the final layer which basically gives you a prediction map in the range [0,1]. But you're comparing it directly with the label map which only contains 2 values 0 and 1. How is it working without rounding?

Moreover, I also noticed you didn't use rounding even during output predictions? Can anyone clarify this a bit?

**Comment by jocicmarko** [comment]
> The question I have is that you're using sigmoid as the activation function in the final layer which basically gives you a prediction map in the range [0,1]. But you're comparing it directly with the label map which only contains 2 values 0 and 1. How is it working without rounding?

There is no "comparison" in the Dice loss function - the computed value of the loss function has a real value which is then backpropagated to train the network.

> Moreover, I also noticed you didn't use rounding even during output predictions? Can anyone clarify this a bit?

You can round them if you want... But I like looking at the real valued predictions, because that way you can see the parts where the models is certain and uncertain about it's predictions.
